configuration/src/main/java/com/synopsys/integration/configuration/config/PropertyConfiguration.java

```diff
@@ -60,6 +63,31 @@ public PropertyConfiguration(@NotNull final List<PropertySource> orderedProperty
         this.orderedPropertySources = orderedPropertySources;
     }
 
+    public List<PropertyInfo> collectPropertyInfo(List<Property> knownProperties, boolean maskPasswords) {
+        List<Property> sortedProperties = sortProperties(knownProperties);
+        List<PropertyInfo> propertyValues = new LinkedList<>();
+        for (Property property : sortedProperties) {
+            if (!wasKeyProvided(property.getKey())) {
+                continue;
+            }
+
+            String value = getRaw(property).orElse("");
+            boolean containsPassword = property.getKey().toLowerCase().contains("password") || property.getKey().toLowerCase().contains("api.token") || property.getKey().toLowerCase().contains("access.token");
+            String maskedValue = value;
+            if (containsPassword && maskPasswords) {
+                maskedValue = StringUtils.repeat('*', maskedValue.length());
+            }
+            propertyValues.add(new PropertyInfo(property.getKey(), maskedValue, property));
+        }
+        return propertyValues;
+    }
+
+    public List<Property> sortProperties(List<Property> knownProperties) {
```

Comment:
I would argue this belongs in a helper utility like PropertyUtils.sortPropertiesByKey.

Reply:
would PropertyConfigUtils be an inappropriate home?

Reply:
That would be fine by me.